File: algorithms/ar-lsrh/golang/linear_search.go
package main

import "log"

// linearSearch linearly scans arr to check for existence of num.
// If num exists then it returns the index of the first match.
// If the number does not exist it returns -1.
func linearSearch(arr []int, num int) int {
	for i, v := range arr {
		if v == num {
			return i
		}
	}

	return -1
}

// go run linear_search.go
func main() {
	log.Print(linearSearch(
		[]int{10, -1, 23, 454, 64, 124, 51},
		124,
	))
}

File: algorithms/ar-lsrh/python2/linear_search.py
def linear_search(arr, value):
    """
    :param arr: array/list to search a value in
    :param value: value to be searched in the array/list
    :return: index -> returns -1 if value not found else return the index of the first occurrence of the value in arr
    """
    for index, item in enumerate(arr):
        if item == value:
            return index

    return -1


def main():
    arr = [2, 3, 0, 4]
    val = 3
    index = linear_search(arr, val)
    print index


if __name__ == '__main__':
    main()
